dsnp-wallet/dsnp-wallet/Source/ViewControllers/ProfileViewController.swift

```diff
@@ -5,10 +5,366 @@
+
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.distribution = .fill
+
+        embeddedView.addSubview(stackView)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.leadingAnchor.constraint(equalTo: embeddedView.leadingAnchor).isActive = true
+        stackView.centerXAnchor.constraint(equalTo: embeddedView.centerXAnchor).isActive = true
+        stackView.topAnchor.constraint(equalTo: embeddedView.layoutMarginsGuide.topAnchor, constant: 10).isActive = true
+        stackView.bottomAnchor.constraint(equalTo: embeddedView.layoutMarginsGuide.bottomAnchor).isActive = true
+        
+        return stackView
+    }
+    
+    private func getTextfieldViewWith(titleLabel: String, placeholder: String) -> UIView {
+        let stackview = UIStackView()
+        stackview.distribution = .fill
+        stackview.axis = .vertical
+        
+        let label = getLabel(title: titleLabel)
+        let textfield = getTextField(with: placeholder)
+        switch titleLabel {
+        case profileText:
+            self.emailTextfield = textfield
+        case bioText:
+            self.bioTextfield = textfield
+        default:
+            return UIView()
+        }
+        
+        let view = UIView()
+        view.addSubview(stackview)
+        stackview.translatesAutoresizingMaskIntoConstraints = false
+        stackview.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: CGFloat(leadingConstraintConstant)).isActive = true
+        stackview.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        stackview.topAnchor.constraint(equalTo: view.topAnchor, constant: 10).isActive = true
+        stackview.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
+        
+        stackview.addArrangedSubview(label)
+        stackview.addArrangedSubview(SharedSpacer(height: 8))
+        stackview.addArrangedSubview(textfield)
+    
+        return view
+    }
+    
+    private func getLabel(title: String) -> UILabel {
+        let label = UILabel()
+        label.text = title
+        label.font = UIFont.Theme.semibold(ofSize: 15)
+        label.textColor = UIColor.Theme.accentBlue
+        
+        return label
+    }
+    
+    private func getTextField(with placeholder: String) -> UITextField {
+        let textfield = UITextField()
+        textfield.backgroundColor = .white
+        textfield.delegate = self
+        textfield.placeholder = placeholder
+        textfield.font = UIFont.Theme.regular(ofSize: 12)
+        textfield.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        textfield.layer.cornerRadius = 4
+        textfield.leftView = UIView(frame: CGRect(x: 0, y: 0, width: 15, height: textfield.frame.height)) //padding
+        textfield.leftViewMode = .always
+        
+        return textfield
+    }
+    
+    private func getPermissionsView() -> UIView {
```

Comment:
Can we make this a standalone class? That way we can more easily move it around the app. I think this is likely to be separated from profile in the long run so it'll be good to make this more adaptable now.